backend/hubs/serializers.py

```diff
@@ -0,0 +1,160 @@
+from rest_framework import serializers
+from django.contrib.auth.models import User
+from .models import Hub
+
+#Serializer for a Hub model with all fields included.
+#This serializer represents a hub
+class HubSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Hub
+        fields = '__all__'
+
+#Serializer for a handling making a new hub.
+# Hub = {"name": "MY NEW HUB", "description": "A Cool Hub"}
+class HubCreateSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Hub
+        fields = ['name', 'description']
+
+    def create(self, validated_data):
+        request = self.context.get('request')
+        user = request.user
+        newly_created_hub = Hub.objects.create(owner=user, **validated_data)
+        return newly_created_hub
+
+
+#Serializer for updating a hub.
+# Hub = {"name": "NEW NAME", "description": "NEW DESC"}
+# we check the request to see if caller is owner or a moderator. 
+class HubUpdateSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Hub
+        fields = ['name', 'description'] 
+
+    def update(self, instance, validated_data):
+        request = self.context.get('request')
+        user = request.user
+        if instance.owner == user or user in instance.mods:
+            instance.name = validated_data.get('name', instance.name)
+            instance.description = validated_data.get('description', instance.description)
+            instance.save()
+            return instance
+        else:
+            raise serializers.ValidationError("Cannot Update Hub : You Are Not Owner/Moderator Of Hub")
```

Comment:
Will the user see this message, or should we just send a error code so that the frontend can decide what message to display

Reply:
The user won't see this, this is the message returned from making an invalid call. we can handle what the user sees on the front end. the error code will be a 400 bad request